plugin/core/src/main/groovy/com/novoda/gradle/release/internal/AndroidAttachments.groovy

```diff
@@ -51,13 +51,22 @@ class AndroidAttachments extends MavenPublicationAttachments {
 
     private static Task androidJavadocsJarTask(Project project, String publicationName, def variant) {
         Javadoc javadoc = project.task("javadoc${publicationName.capitalize()}", type: Javadoc) { Javadoc javadoc ->
-            javadoc.source = variant.javaCompiler.source
-            javadoc.classpath = variant.javaCompiler.classpath
+            if(variant.hasProperty('javaCompileProvider')) {
+                javadoc.source = variant.javaCompileProvider.get().source
```

Comment:
I would prefer to extract a little method where to put this if statement (like we have already done in [another plugin](https://github.com/novoda/gradle-static-analysis-plugin/blame/master/plugin/src/main/groovy/com/novoda/staticanalysis/internal/CodeQualityConfigurator.groovy#L133-L139)) so we can remove some duplication